app/models/cloud_cluster.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+require "velum/salt"
+# CloudCluster represents user-configured attributes of a cloud deployment.
+class CloudCluster
+  include ActiveModel::Model
+  attr_accessor :cloud_framework,
+    :instance_count, :instance_type, :instance_type_custom,
+    :subnet_id, :security_group_id # EC2
```

Comment:
do you think it makes sense to abstract the ec2 part out of this class into it's own mode, and use `CloudCluster` as a base class? i just imaging when adding more cloud providers this might be more organized in the long term, as (at least) the attributes will probably differ

Reply:
I've been looking at both ways... I know we have a lot of duplication across all the providers we will support - I'm not going to be certain until I add Azure support. So, I'd like to leave it as-is, and make that decision when I hit the problem... for now that's just YAGNI.